plugins/nlp/nlp_eng.cpp

```diff
@@ -0,0 +1,79 @@
+/*##############################################################################
+1
+    HPCC SYSTEMS software Copyright (C) 2021 HPCC Systems�.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#include "nlp_eng.hpp"
+#include "nlp_engine.h"
+
+static NLP_ENGINE *nlpEngine = NULL;
+
+NLPEng::NLPEng(const char *workingFolderPath) {
+    if (workingFolderPath[0]) {
+        workingFolder = (char *)malloc(strlen(workingFolderPath)+1);
```

Comment:
Unless I'm misreading this is guaranteed not to be long enough. Again, VStringBuffer would be simple and safe.